lib/grpcServer/handlers/core/updateStateHandlerFactory.js

```diff
@@ -0,0 +1,63 @@
+const cbor = require('cbor');
+const {
+  UpdateStateTransitionResponse,
+} = require('@dashevo/dapi-grpc');
+
+const InvalidArgumentGrpcError = require('../../error/InvalidArgumentGrpcError');
+const InternalGrpcError = require('../../error/InternalGrpcError');
+
+/**
+ *
+ * @param {jaysonClient} rpcClient
+ * @returns {updateStateHandler}
+ */
+function updateStateHandlerFactory(rpcClient) {
+  /**
+   * @typedef updateStateHandler
+   * @param {Object} call
+   */
+  async function updateStateHandler(call) {
+    const { request } = call;
+    const header = request.getHeader();
```

Comment:
Why not just serialize request to base64 and pass to Tendermint? We have such validation in Machine checkTx